skt/__init__.py

```diff
@@ -479,16 +480,38 @@ def __init__(self, path, basecfg, cfgtype=None, makeopts=None,
         self.basecfg = os.path.expanduser(basecfg)
         self.cfgtype = cfgtype if cfgtype is not None else "olddefconfig"
         self._ready = 0
-        self.makeopts = None
         self.buildlog = "%s/build.log" % self.path
         self.defmakeargs = ["make", "-C", self.path]
         self.enable_debuginfo = enable_debuginfo
 
         if makeopts is not None:
-            # FIXME: Might want something a bit smarter here, something that
-            # would parse it the same way bash does
-            self.makeopts = makeopts.split(' ')
-            self.defmakeargs += self.makeopts
+            self.makeopts = self.defmakeargs + shlex.split(makeopts)
+        else:
+            self.makeopts = self.defmakeargs
```

Comment:
This is getting too confusing due to several reasons. First, we assign `self.defmakeargs` and then never use it outside the constructor. Then we have the `makeopts` argument, and we have `self.makeopts`, but they're not the same. The argument is named `makeopts`, but it's not just "options", it's any arguments. Lastly, `self.makeopts` is not just "options", neither it's just "arguments", because it also contains `argv[0]`, the executable name (`"make"`), which, strictly speaking, is not an argument, as in most programs don't process it. Same goes to naming them `args` throughout the code, because those also include `argv[0]`.

With that in mind, how about this:
* rename the constructor argument to `make_args` (or similar, or perhaps even `extra_make_args`) and submit an issue to rename the command-line option accordingly later;
* remove `self.defmakeargs`;
* use `self.make_argv` (or similar, or even `self.make_argv_base`, or `self.make_argv_pfx`) to signify that these are more than just arguments (in the widely-accepted sense);
* use `argv` instead of `args` when building the make's argv array before execution.

I think it is better to name things as close as possible to their contents to minimize the cognitive load and confusion when reading the code, and that's what I'm targeting above.